cmd/acraserver/listener.go

```diff
@@ -152,12 +277,61 @@ func (server *SServer) handleCommandsConnection(connection net.Conn) {
 }
 
 // start listening commands connections from proxy
-func (server *SServer) StartCommands() {
-	listener, err := network.Listen(server.config.GetAcraAPIConnectionString())
+func (server *SServer) StartCommands(graceful bool) {
+	var listener net.Listener
+	var listenerGraceful *net.TCPListener
+	var err error
+	var connection net.Conn
+	if graceful == true {
+		listenerGraceful, err = network.ListenTCP(server.config.GetAcraAPIConnectionString())
```

Comment:
You can use this function - https://github.com/cossacklabs/acra/blob/master/network/utils.go#L22 to support unix/tcp

Reply:
Instead of having `if` checks everywhere, can we create two separate functions, one for unix sockets, another for TCP, and use `if` once:

```
func (server *SServer) StartCommands(graceful bool) {

...

	if graceful == true {
	   handleGracefulSocket()
	} else {
	   handleSocket() 
	}
}

func handleGracefulSocket()
func handleSocket()
```

Reply:
Refactored